### PR TITLE
Das/seed data

### PR DESCRIPTION
**Pull request by Woonyong92**
@junwen29 @jkcheong92 
Finish seeding of data.
Do note, when running seed file, you have to reset your database.
Also do note that as there is a lot of data to be seeded, it will data some time

### CODE REVIEW

**Comment by Woonyong92** [comment]
@jkcheong92 @junwen29  reopened the pull request with the bug fixed. Realised the bug only when i tried the seeded data in the develop branch

File: db/seeds.rb [hunk 1]
```diff
@@ -11,11 +9,787 @@
...
+
+start_date = '01-12-2014'.to_date
+end_date = start_date + 1.months
+while merchant_id < 1050
+  Payment.new(id: payment_id, start_date: start_date, expiry_date: end_date, total_cost: '40.00', add_on1: true, add_on2: true, add_on3: false, plan1: true, paid: true, created_at: start_date, updated_at: '2015-10-01 00:00:00', merchant_id: merchant_id, months: 1).save(validate: false)
+  PlanPayment.create(id: plan_payment_id, plan_id: 1, payment_id: payment_id, created_at: start_date, updated_at: '2015-10-01 00:00:00')
+  AddOnPayment.create(id: add_on_payment_id, add_on_id: 1, payment_id: payment_id, created_at: start_date, updated_at: '2015-10-01 00:00:00')
+  add_on_payment_id = add_on_payment_id + 1
+  AddOnPayment.create(id: add_on_payment_id, add_on_id: 2, payment_id: payment_id, created_at: start_date, updated_at: '2015-10-01 00:00:00')
+  if merchant_id%rand(5..10) == 0
```
Comment (jkcheong92) [inline db/seeds.rb]: what is this rand for?

File: db/seeds.rb [hunk 2]
```diff
@@ -11,11 +9,787 @@
...
+end
+
+# Seed DealAnalytics for View count for deal 1000
+view_count = 1000
+start_date = DateTime.parse("2015-09-20 00:00:00")
+end_date = DateTime.parse("2015-10-20 00:00:00")
+while start_date <= end_date
+  limit = view_count + rand(20..40)
+  while view_count < limit
+    Viewcount.create(id: view_count, deal_id: 1000, user_id: view_count, created_at: start_date)
```
Comment (jkcheong92) [inline db/seeds.rb]: Do we have user_id up to 1000+, cos view_count is going to get super large

File: db/seeds.rb [hunk 3]
```diff
@@ -11,11 +9,787 @@
...
+
+# Seed DealAnalytics for View count for deal 1001
+redemption = 1000
+start_date = DateTime.parse("2015-10-01 00:00:00")
+end_date = DateTime.parse("2015-10-20 00:00:00")
+while start_date <= end_date
+  limit = redemption + rand(rand(10..30)..rand(60..100))
+  venue_id = rand(1000..1002)
+  while redemption < limit
+    Redemption.create(id: redemption, deal_id: 1001, user_id: redemption, venue_id: venue_id, created_at: start_date)
```
Comment (jkcheong92) [inline db/seeds.rb]: Same thing, do we have that many user_id?

File: db/seeds.rb [hunk 4]
```diff
@@ -11,11 +9,787 @@
...
+while start_date <= end_date
+  limit = view_count + rand(rand(50..90)..rand(100..150))
+  while view_count < limit
+    Viewcount.create(id: view_count, deal_id: 1001, user_id: view_count, created_at: start_date)
+    view_count = view_count + 1
+  end
+  start_date = start_date + 1
+end
+num_view_count = view_count - starting_view_count
+unique_view_count = num_view_count- rand(50..90)
```
Comment (jkcheong92) [inline db/seeds.rb]: Do we still need to update DealAnalytic now? Cos the method to set view count has already been implemented

File: db/seeds.rb [hunk 5]
```diff
@@ -11,11 +9,787 @@
...
+  limit = view_count + rand(20..40)
+  while view_count < limit
+    Viewcount.create(id: view_count, deal_id: 1011, user_id: view_count, created_at: start_date)
+    view_count = view_count + 1
+  end
+  start_date = start_date + 1
+end
+num_view_count = view_count - starting_view_count
+unique_view_count = num_view_count- rand(50..90)
+DealAnalytic.create(id: 1005, deal_id: 1011, view_count: num_view_count, unique_view_count: unique_view_count, redemption_count: num_redemption)
```
Comment (jkcheong92) [inline db/seeds.rb]: No deal analytic created for deal_id: 1005 - 1010?

File: db/seeds.rb (hunk 1 above)
Comment (Woonyong92) [inline db/seeds.rb]: this is for changing to the next month. If I do not put rand then the number of subscribers will be fixed each month making the line chart in payment analytics a straight line

File: db/seeds.rb (hunk 2 above)
Comment (Woonyong92) [inline db/seeds.rb]: Oh, I didn't seed users yet. But it is not necessary to have actual users so it is ok

File: db/seeds.rb (hunk 3 above)
Comment (Woonyong92) [inline db/seeds.rb]: Same as before it is not necessary to have actual users so it is ok

File: db/seeds.rb (hunk 4 above)
Comment (Woonyong92) [inline db/seeds.rb]: Maybe it is not necessary but we can test later first. Also, I am not sure if unique view count can be set. If so, I might need to change the user id for seeding view counts

File: db/seeds.rb (hunk 5 above)
Comment (Woonyong92) [inline db/seeds.rb]: Yup! 1005 and 1006 are waiting deals. The rest are not really required for the data

File: db/seeds.rb (hunk 2 above)
Comment (jkcheong92) [inline db/seeds.rb]: On admin, we will see the user on the redemption table - so need to seed user too

Comment (Woonyong92) [inline db/seeds.rb]: omg ok...this makes things more complicated...i'll seed the users now

**Pull request merged**